app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
```

Comment:
As mentioned below, I'd keep to just..

   * `angularForm`
   * `saveable` (though I'd make that optinal, since most of the time this will be just `angularForm.$valid && angularForm.$dirty`, so we can default to `miqService.saveable` unless provided)
   * onSave, onCancel, onReset
   * `newRecord`

Handling redirection and flash messages is not the job for a buttons component.